src/nvim/drawline.c

```diff
@@ -567,33 +575,28 @@ static void handle_lnum_col(win_T *wp, winlinevars_T *wlv, int sign_num_attr, in
       && !((has_cpo_n && !wp->w_p_bri) && wp->w_skipcol > 0 && wlv->lnum == wp->w_topline)) {
     // If 'signcolumn' is set to 'number' and a sign is present in "lnum",
     // then display the sign instead of the line number.
-    if (wp->w_minscwidth == SCL_NUM && wlv->sattrs[0].text) {
-      get_sign_display_info(true, wp, wlv, 0, sign_cul_attr);
+    if (wp->w_minscwidth == SCL_NUM && wlv->sattrs[0].text[0]) {
```

Comment:
The `!nrcol` assertion above above will fail if a wrapped line has a sign.
```suggestion
    if (wp->w_minscwidth == SCL_NUM && wlv->sattrs[0].text[0]
        && wlv->row == wlv->startrow + wlv->filler_lines && wlv->filler_todo <= 0) {
```

Reply:
This is already inside an `if (wlv->row == wlv->startrow + wlv->filler_lines)`, so at best the `wlv->filler_todo <= 0` is needed?

Reply:
You missed the `!has_cpo_n`

Reply:
However, "empty" should imply drawing the underlying Number highlight, not opening a gap to the void of the universe. Updating the test.